rear_camera_py/rear_camera/line_angle_calculator.py

```diff
@@ -226,162 +138,129 @@ def get_transformed_image(self, img: np.ndarray, borderValue=(255, 255, 255)) ->
+        canny_th1: int = 50,
+        canny_th2: int = 50,
+        canny_aperture_size: int = 7,  # 3~7
+        do_merge: bool = False
+    ) -> Union[np.ndarray, None]:
+        """画像上の線分を検出する.
 
         Args:
-            img_x (Union[int, float]): 画像上のX座標[pix].
-            img_y (Union[int, float]): 画像上のY座標[pix].
-            img_h (int): 画像の縦幅[pix].
-            img_w (int): 画像の横幅[pix].
+            img (np.ndarray): 射影変換後の画像.
+            dis_from_edge_threshold (int): 射影変換によってできた元画像の枠(直線)と同一とみなす距離
+
+            NOTE:以下の引数の詳細: https://emotionexplorer.blog.fc2.com/blog-entry-128.html
+                                  https://nsr-9.hatenablog.jp/entry/2021/08/12/200000
+            length_threshold (int): 長さ閾値(これより短い線は除外)
+            distance_threshold (float): 距離閾値(この値より遠い座標は、同一線ではない)
+            canny_th1 (int): Cannyヒステリシス1
+            canny_th2 (int): Cannyヒステリシス2
+            canny_aperture_size (int): Cannyソベルオペレータ アパチャーサイズ
+            do_merge (bool): 増分マージオプション
 
         Returns:
-            Tuple[float, float]: 変換後の座標(x, y)[mm].
+            Union[np.ndarray, None]: 検出した線分、線分を検出できなかった場合は`None`を返す.
         """
-        distance_between_aruco_markers_center_and_runner_center_mm = 312.5
-        img_x_axis_and_runner_x_axis_distance_mm = (
-            self.pix_to_mm(self.__height_offset_from_center) +
-            distance_between_aruco_markers_center_and_runner_center_mm
+        # 線分検出モジュールのインスタンス化
+        fld = cv2.ximgproc.createFastLineDetector(
+            length_threshold,
+            distance_threshold,
+            canny_th1,
+            canny_th2,
+            canny_aperture_size,
+            do_merge
         )
-        img_h_mm = self.pix_to_mm(img_h)
-        img_w_mm = self.pix_to_mm(img_w)
-        runner_base_x_mm = self.pix_to_mm(img_x) - img_w_mm/2
-        runner_base_y_mm = img_x_axis_and_runner_x_axis_distance_mm - \
-            (img_h_mm - self.pix_to_mm(img_y))
-        return runner_base_x_mm, runner_base_y_mm
-
-    def runner_base_coordinate_to_image_base_coordinate_pix(
-        self,
-        rx: Union[int, float],
-        ry: Union[int, float],
-        img_h: int,
-        img_w: int
-    ) -> Tuple[int, int]:
-        """機体の中心を原点とした座標に変換する.
 
-        Args:
-            rx (Union[int, float]): 走行体の中心を原点としたX座標[mm].
-            ry (Union[int, float]): 走行体の中心を原点としたY座標[mm].
-            img_h (int): 画像の縦幅[pix].
-            img_w (int): 画像の横幅[pix].
+        # 線分の2点座標を検出　※型:[[[x1,y1,x2,y2]],...]
+        lines = fld.detect(img)
 
-        Returns:
-            Tuple[int, int]: 変換後の座標(x, y)[pix].
-        """
-        pix_x = self.mm_to_pix(rx) + img_w/2
-        pix_y = self.__height_offset_from_center + 312.5 - self.mm_to_pix(ry)
-        return pix_x, pix_y
+        # 画像上側7.5割の領域上にある線、もしくは、またがってる線分を削除
+        ignore_border = int(img.shape[0] * 0.75)
+        lines = lines[np.where((lines[:, :, 1] > ignore_border) & (lines[:, :, 3] > ignore_border))]
```

Comment:
`self.runner_base_coordinate_to_image_base_coordinate_pix`関数を使うと、走行体の中心からの座標[mm]を画像上の座標[pix]に変換できるので、検出領域を変更したくなった時の指定が少し楽になるかも？

あと、射影変換のパラメータの変更等で射影変換の際、元画像を拡大するようにパラメータを変更した際などの、影響を吸収できるようになる。
ただ、カメラの角度が変わるだけでは何の影響もないので、余裕がない場合はこのままでもいい気がします。

以下、コード例（未検証）
```suggestion
        # 走行体の中心から600mmより上にある線、もしくは、またがってる線分を削除
        img_h, img_w= img.shape[:2]
        _, ignore_border = self.runner_base_coordinate_to_image_base_coordinate_pix(0, 600, img_h, img_w)
        lines = lines[np.where((lines[:, :, 1] > ignore_border) & (lines[:, :, 3] > ignore_border))]
```


同様に、`self.mm_to_pix`関数を利用すると引数の`length_threshold`とかの単位を`mm`で指定できます。

以下、走行体を基準にした座標と画像座標の関係。
![basic-coordinates](https://user-images.githubusercontent.com/31930148/200966845-235cccda-9320-4c43-8122-cb94a63e587e.png)